unified_ecommerce/test_utils.py

```diff
@@ -138,3 +223,211 @@ class PickleableMock(Mock):
+            assert_json_equal(
+                *make_timestamps_matchable([response.data, dj_serializer])
+            )
+
+        if isLoggedIn and instance.is_active is not None:
+            # Deleted items should return a 404. (is_active is None if it's active;
+            # otherwise it has a timestamp of when it was deleted.)
+            assert response.status_code == 404
+
+    def test_update(self, update_data, isLoggedIn, client, user_client):
+        """
+        Test that the viewset can update an object.
+
+        Args:
+        - update_data (dict): the data to use for the update
+        - isLoggedIn (bool): whether or not the client is logged in
+        - client (APIClient): the client to use for non-logged in requests
+        - user_client (APIClient): the client to use for logged in requests
+
+        Returns:
+        - tuple of (instance, response): the instance that was updated and the response
+        """
+        instance = self.factory_class()
+
+        use_client = user_client if isLoggedIn else client
+        response = use_client.patch(
+            self.object_url.format(instance.pk), data=update_data
+        )
+
+        if not isLoggedIn:
+            assert response.status_code == 403
+        return (instance, response)
+
+    def test_delete(self, isLoggedIn, client, user_client):
+        """
+        Test that the viewset can delete an object. Note that this will actually test
+        deletion.
+
+        Args:
+        - isLoggedIn (bool): whether or not the client is logged in
+        - client (APIClient): the client to use for non-logged in requests
+        - user_client (APIClient): the client to use for logged in requests
+
+        Returns:
+        - tuple of (instance, response): the instance that was deleted and the response
+        """
+        instance = self.factory_class()
+
+        use_client = user_client if isLoggedIn else client
+        response = use_client.delete(self.object_url.format(instance.pk))
+
+        assert response.status_code == 403 if not isLoggedIn else 204
+
+        if isLoggedIn:
+            assert self.queryset.filter(pk=instance.pk).count() == 0
+
+        return (instance, response)
+
+    def test_create(self, create_data, isLoggedIn, client, user_client):
+        """
+        Test that the viewset can create an object.
+
+        Args:
+        - create_data (dict): the data to use for the update
+        - isLoggedIn (bool): whether or not the client is logged in
+        - client (APIClient): the client to use for non-logged in requests
+        - user_client (APIClient): the client to use for logged in requests
+
+        Returns:
+        - response (Response): the response from the API
+        """
+        use_client = user_client if isLoggedIn else client
+        response = use_client.post(self.list_url, data=create_data)
+
+        if not isLoggedIn:
+            assert response.status_code == 403
```

Comment:
Should this assert a 201 in an `else` to be sure a 500 error isn't a passing test?